Tests/PHPUnit/Components/ApplePayDirect/Services/ApplePayButtonBuilderTest.php

```diff
@@ -0,0 +1,419 @@
+<?php
+
+namespace MollieShopware\Tests\Components\ApplePayDirect\Services;
+
+use Enlight_Controller_Request_Request;
+use Enlight_Template_Manager;
+use Enlight_View_Default;
+use Exception;
+use MollieShopware\Components\Account\Account;
+use MollieShopware\Components\ApplePayDirect\Services\ApplePayButtonBuilder;
+use MollieShopware\Components\ApplePayDirect\Services\ApplePayDirectDisplayOptions;
+use MollieShopware\Components\ApplePayDirect\Services\ApplePayPaymentMethod;
+use MollieShopware\Components\Config;
+use MollieShopware\Tests\PHPUnit\Utils\Fakes\View\FakeView;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use sAdmin;
+use sBasket;
+use Shopware\Models\Shop\Currency;
+use Shopware\Models\Shop\Shop;
+use Shopware_Components_Config;
+
+class ApplePayButtonBuilderTest extends TestCase
+{
+    /**
+     * @var Account|MockObject
+     */
+    private $accountService;
+
+    /**
+     * @var Config
+     */
+    private $mollieConfig;
+
+    /**
+     * @var ApplePayPaymentMethod|MockObject
+     */
+    private $applePayPaymentMethod;
+
+    /**
+     * @var ApplePayDirectDisplayOptions|MockObject
+     */
+    private $restrictionService;
+
+    /**
+     * @var MockObject|sBasket
+     */
+    private $sBasket;
+
+    /**
+     * @var Enlight_View_Default
+     */
+    private $view;
+
+    /**
+     * @var Enlight_Controller_Request_Request
+     */
+    private $request;
+
+    /**
+     * @var MockObject|Shop
+     */
+    private $shop;
+
+    /**
+     * @var ApplePayButtonBuilder
+     */
+    private $buttonBuilder;
+
+    public function setUp(): void
+    {
+        $this->accountService = $this->createMock(Account::class);
+        $this->mollieConfig = $this->createMock(Config::class);
+        $this->sBasket = $this->createMock(sBasket::class);
+        $this->view = new FakeView($this->createMock(Enlight_Template_Manager::class));
+        $this->request = $this->createMock(Enlight_Controller_Request_Request::class);
+
+        $this->setUpShop();
+        $this->setUpRestrictionService();
+        $this->setUpApplePayPaymentMethod();
+
+        $this->buttonBuilder = new ApplePayButtonBuilder(
+            $this->accountService,
+            $this->mollieConfig,
+            $this->createMock(Shopware_Components_Config::class),
+            $this->applePayPaymentMethod,
+            $this->restrictionService,
+            $this->getAdminModuleMock(),
+            $this->sBasket
+        );
+    }
+
+    /**
+     * @test
+     * @testdox Apple Pay direct button is not active when restricted by risk management.
+     *
+     * @throws Exception
+     * @return void
+     */
+    public function testIfApplePayButtonNotActiveWhenRestrictedByRiskManagement()
```

Comment:
i really really like thos tests
still,  please find way way shorter names, it's nearly impossible to figure out existing cases when just looking at it, so it can indeed happen that a test exists twice

Reply:
I feel like being specific is quite important in tests. But I'll try to shorten them.

Reply:
and i didnt say sth different, specific and short if possible
tests need to cover both things, being specific and easy readable